ob-chatgpt-shell.el

```diff
@@ -65,18 +65,29 @@ This function is called by `org-babel-execute-src-block'"
         (cons 'role "system")
         (cons 'content (map-elt params :system)))))
     (when (map-elt params :context)
-      (ob-chatgpt-shell--context))
+      (ob-chatgpt-shell--context (map-elt params :context)))
     `(((role . "user")
        (content . ,body))))
    (map-elt params :version)
    nil nil
    (map-elt params :temperature)))
 
-(defun ob-chatgpt-shell--context ()
-  "Return the context (what was asked and responded) in all previous blocks."
+(defun ob-chatgpt-shell--context (context-name)
+  "Return the context (what was asked and responded) for matching
+previous src blocks. Each must have a :context source block arg
+with a value matching `CONTEXT-NAME'."
   (let ((context '()))
     (mapc
      (lambda (src-block)
+       (let ((system (map-elt (map-elt src-block 'parameters '()) :system)))
+         (when system
+           (if context
```

Comment:
If I'm understanding correctly, `if context` check is impliying that if there's a context there must be an existing system role, which may not always be the case. How about explictly checking for an existing system role?

For example

```
(if (seq-some
     (lambda (x)
       (equal (map-elt x 'role) "system"))
     context)
    ...
  ...)
```

Alternatively, we can keep a local var context-system